addons/partner_autocomplete/models/res_company.py

```diff
@@ -12,3 +22,90 @@ class ResCompany(models.Model):
     def _inverse_partner_gid(self):
         for company in self:
             company.partner_id.partner_gid = company.partner_gid
+
+    def _enrich(self):
+        """
+        This method calls the partner autocomplete service from IAP to enrich the partner related fields
+        of the company.
+        """
+        self.ensure_one()
+
+        company_domain = self._get_company_domain()
+        if not company_domain:
+            return {'error': _("Could not extract company domain. Impossible to enrich the company.")}
+
+        company_data = self.env['res.partner'].enrich_company(company_domain, False, self.vat)
+        if company_data.get('error'):
+            return {'error': company_data.get('error_message', _('Unkown reason'))}
+
+        additional_data = company_data.pop('additional_info', False)
+
+        if company_data.get('logo'):
+            # Erase image_1920 even if something is in it. Indeed as partner_autocomplete is probably installed as a
+            # core app (mail -> iap -> partner_autocomplete auto install chain) it is unlikely that people already
+            # updated their company logo.
+            try:
+                company_data['image_1920'] = base64.b64encode(requests.get(company_data['logo']).content)
```

Comment:
Please be sure that all calls to external systems are configured with a short timeout, typically 5 seconds, because it blocks the whole HTTP worker that is executing it. IAP calls normally have their own default timeouts, which should be fine.